### DISCUSSION TITLE
Run the LibreOffice renderer on the client

### DISCUSSION THREAD

**Discussion by Thesola10** [CollaboraOnline/online#7270] [Ideas]
Would it be possible, in the long term, to re-architecture Collabora Online to be able to run the LibreOffice backend process (`kit`, if I'm not mistaken) in Emscripten, on the client device directly? This would address latency and bandwidth issues with the current remote rendering approach, which make Collabora Online borderline unusable over suboptimal Internet connections.

Ideally the WASM module would be cached on the user's browser, not unlike a tiny, portable install of LibreOffice headless in the tab. Given that Collabora Online doesn't appear to require any of LibreOffice's frontend code, it might not be too far-fetched to think the LibreOffice instance itself could run on the client, bringing it on par with OnlyOffice or Google Docs in terms of performance.

Of course, I understand the issues this could raise with collaborative editing, but on the flip side, solving that aspect would bring collaborative editing to desktop releases of LibreOffice as well, making for a much more compelling product.

**Comment by lokdoc** [comment]
Interesting , Same question

**Comment by tockudex** [comment]
Do we have some news on this possibility?
